syntax = "proto3";

package mutr;

// Java 관련 옵션 설정
option java_multiple_files = true; // 메시지와 서비스를 개별 파일로 분리하여 인식률 향상
option java_package = "com.teno.mutr.node.infra.grpc.stubs"; // 생성될 자바 패키지 경로 지정
option java_outer_classname = "AnalysisProto"; // 바깥 클래스 이름 명시 (선택 사항)

service AnalysisService {
  rpc AnalyzeNode (AnalysisRequest) returns (AnalysisResponse);
}

message AnalysisRequest {
  string content = 1;          // 현재 노드 글
  string parent_summary = 2;   // 이전 노드 요약 (변조용)
  string full_context = 3;     // 이전까지의 전체 글 (요약용)
}

message AnalysisResponse {
  string topic = 1;
  string emotion = 2;
  float confidence = 3;
  float mutation_score = 4;
}